packages/parser/lib/src/parser/grammar.dart

```diff
@@ -600,13 +600,14 @@ class ParserGrammar {
       // bitwise not
       ..leftExpression(LexerGrammar.BAR)
       // type check
-      ..leftComplex<OperatorToken, Expression>(
-        (LexerGrammar.WS &
+      ..postfix<List<dynamic>, Expression>(
+        LexerGrammar.NLs &
                 (LexerGrammar.AS | LexerGrammar.AS_SAFE) &
-                LexerGrammar.WS)
-            .map((it) => it[1] as OperatorToken),
-        mapper: (left, operator, right) =>
-            BinaryExpression(_id++, left, operator, right),
+                LexerGrammar.NLs &
```

Comment:
Are you sure that `NLs` works correctly? `WS` requires some whitespace, whereas `NLs` only allows some optional whitespace.